sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1866,34 +1866,46 @@ void ProgramManager::addImages(sycl_device_binaries DeviceBinary) {
 
     m_BinImg2KernelIDs[Img.get()].reset(new std::vector<kernel_id>);
 
-    for (sycl_offload_entry EntriesIt = EntriesB; EntriesIt != EntriesE;
-         ++EntriesIt) {
+    sycl_offload_entry EntriesIt = EntriesB;
+    // Assuming that there isn't a mix of new and old offload entry types in the
```

Comment:
Not sure if this is an assumption. I suppose it is just not supported. May be reword: A mix of old and new offload entry types is not supported.

Thanks